homeassistant/components/zwave_js/config_flow.py

```diff
@@ -901,12 +1008,61 @@ async def async_step_configure_addon(
 
         return self.async_show_form(step_id="configure_addon", data_schema=data_schema)
 
+    async def async_step_choose_serial_port(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Choose a serial port."""
+        if user_input is not None:
+            addon_info = await self._async_get_addon_info()
+            addon_config = addon_info.options
+            self.usb_path = user_input[CONF_USB_PATH]
+            new_addon_config = {
+                **addon_config,
+                CONF_ADDON_DEVICE: self.usb_path,
+            }
+            if addon_info.state == AddonState.RUNNING:
+                self.restart_addon = True
+            # Copy the add-on config to keep the objects separate.
+            self.original_addon_config = dict(addon_config)
+            # Remove legacy network_key
+            new_addon_config.pop(CONF_ADDON_NETWORK_KEY, None)
+            await self._async_set_addon_config(new_addon_config)
+            return await self.async_step_start_addon()
+
+        ports = await async_get_usb_ports(self.hass)
```

Comment:
Existing code, but I think we should catch `OSError` here and abort if we can't get USB ports.